modules/cnvkit/main.nf

```diff
@@ -31,13 +31,23 @@ process CNVKIT {
     path "versions.yml"           , emit: versions
 
     script:
+    def target_args = ""
+
+    if (options.args.contains("--method wgs") || options.args.contains("-m wgs")) {
+        target_args = targetfile ? "--targets $targetfile" : ""
+    }
+    else {
+        target_args = "--targets $targetfile"
+    }
+
```

Comment:
wouldn't 

```
target_args = targetfile ? "--targets $targetfile" : ""
```

be enough without the if-clause?

Reply:
The way I read the code, it seems that making the --target argument optional when running as `--method wgs`(which makes sense according to the documentation and it's the [name of the commit](https://github.com/fbdtemme/modules/commit/384994353e63cccfc0672d86e4b1d50ef1b32f5d)). There's also [test for it](https://github.com/fbdtemme/modules/blob/07fa77109dba21ed57119677332e2cac418e65f1/tests/modules/cnvkit/main.nf#L6). If you give that test a targets argument (other than `[]`) it fails. My guess is that it actually should be blank in [line 37](https://github.com/fbdtemme/modules/blob/07fa77109dba21ed57119677332e2cac418e65f1/modules/cnvkit/main.nf#L37), like return a blank line no matter what - because [it's not compatible](https://cnvkit.readthedocs.io/en/stable/nonhybrid.html#whole-genome-sequencing-wgs) to have `wgs` and `targets` together. But it's a guess - I'm actually a bit confused what the intention is here.

Reply:
As I read the documentation it is possible to use a --target argument in wgs mode to limit the analysis to specific regions of interest, but this is an optional parameter. For the hybrid mode the targets is a required input parameter. The intention was to make it possible to pass [] as target when running in wgs mode while still allowing the parameter if required.